src/Datagrid/Datagrid.php

```diff
@@ -293,4 +293,25 @@ public function getForm()
 
         return $this->form;
     }
+
+    /**
+     * Returns the default filter value if the user has not set it.
+     */
+    private function resolveFilterValue(?array $data, string $name, FilterInterface $filter): ?array
+    {
+        $userValue = $data[$filter->getFormName()] ?? null;
+        if (!isset($this->values[$name])) {
+            return $userValue;
+        }
+
+        $withoutNulls = array_filter($userValue ?? [], static function ($val): bool {
```

Comment:
I may don't understand what you're doing, but it seems like you're removing the submitted `null` value.

What if the user is looking for `null` ?

For example, the choiceFilter allow the null value: https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/999

Does it break the feature ?